### ISSUE TITLE
Floating point numbers cause incorrect test results

### ISSUE THREAD

**Issue by ThomasR** [ericam/true#44]
This simple test

    @include assert-equal(.1 + .2, .3, '.3 and .3 should be equal.');

results in the following output:

    DEBUG: [FAIL] Simple assertions » .3 and .3 should be equal.
    DEBUG: Expected number »
    DEBUG: 0.3
    DEBUG: Returned number »
    DEBUG: 0.3
    …
    -result: FAIL;
    -description: ".3 and .3 should be equal.";
    -expected--number: 0.3;
    -returned--number: 0.3;

Which is very confusing, to say the least.

Note that the test fails because `.1 + .2 == .30000000000000004`, which is a [well-known limitation](http://stackoverflow.com/questions/588004/is-floating-point-math-broken) of IEEE floating point arithmetics.

**Expected output**

    DEBUG: [FAIL] Simple assertions » .3 and .3 should be equal.
    DEBUG: Expected number »
    DEBUG: 0.3
    DEBUG: Returned number »
    DEBUG: 0.30000000000000004
    …
    -result: FAIL;
    -description: ".3 and .3 should be equal.";
    -expected--number: 0.3;
    -returned--number: 0.30000000000000004;

Not sure what exactly causes this issue. I think it may be due to [SASS's implementation of Number:inspect](https://github.com/sass/sass/blob/cc96f0c85d5ec09324e9158585c23ddbe922b2ab/lib/sass/script/value/number.rb#L284), which rounds the given number prior to printing.

So the solution here should be either
* circumvent the abovementioned SASS-provided rounding (how? Removing the explicit `inspect()` calls has no effect - I guess SASS calls it internally)
* round numbers before comparing, maybe something like
`@return inspect($one) == inspect($two);` instead of `@return $one == $two;`
in [_utilities.scss](https://github.com/ericam/true/blob/f42b7210488f5df75ca2636da3f82dcdb25715b7/sass/true/_utilities.scss#L58)

**Comment by ericam** [comment]
I'd like input from @chriseppstein, @nex3, or @xzyfer on how something like this is best addressed. It seems to me like a Sass issue, and not something we should have to work around on our end.

**Comment by ThomasR** [comment]
Sass will have to apply rounding at some point.
The problem here is not the rounding error as such, but the mixture of Sass's output and its internal representation of the values.

Maybe this is a better example:
```scss
@include test('division') {
  @include assert('divide') {
    @include input { line-height: (1 / 3.0); }
    @include expect { line-height: 0.33333; }
  } // PASS
  @include assert-equal(1 / 3.0, 0.33333); // FAIL with confusing error message
  @include assert-equal(1 / 3.0, 0.3333333333333333); // PASS
}
```
So, there's basically two parts of the problem:

1. `assert-equal()` behaves differently from `assert()` in that it inspects Sass's intermediate values rather than the output. This is arguably ok, but should at least be well-documented.
2. The error output of `assert-equal()` is bad because printing the values via Sass means that everything will be rounded no matter what.

**Comment by nex3** [comment]
Chris fixed this in sass/sass@cc96f0c85d5ec09324e9158585c23ddbe922b2ab, which has been released as 3.4.19.

**Comment by ericam** [comment]
Thanks @nex3!
